index.bs

```diff
@@ -2222,6 +2267,95 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
 Issue: Invoke [=parse summary buckets=] and [=parse summary window operator=]
 from this algorithm.
 
+To <dfn>parse max number of event states</dfn> from a [=map=] |map|,
+a [=source type=] |sourceType|,
+a possibly null 32-bit unsigned integer |attributionScopeLimit|,
+and a [=randomized response output configuration=] |config|:
+1. If |attributionScopeLimit| is null:
+    1. If |map|["`max_event_states`"] [=map/exists=], return an error.
+    1. Return null.
+1. Let |states| be the [=obtain a set of possible trigger states|number of possible trigger states=] with |config|.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| is greater than [=default max number of event states=]
+    and |map|["`max_event_states`"] does not [=map/exist=], return an error.
+1. Let |maxNumEventStates| be the result of running
+    [=parse an optional 32-bit unsigned integer=] with |map|,
+    "`max_event_states`", and [=default max number of event states=].
+1. If |maxNumEventStates| is an error or |maxNumEventStates| is equal to 0 or |maxNumEventStates| is greater than [=max trigger-state cardinality=], return an error.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| is greater than |maxNumEventStates|, return an error.
+1. Return |maxNumEventStates|.
+
+To <dfn>parse attribution scopes</dfn> from a [=map=] |map| and a possibly null 32-bit unsigned integer |attributionScopeLimit|:
+1. Let |result| be a [=set=].
+1. If |map|["`attribution_scopes`"] [=map/exists=]:
+    1. Let |values| be |map|["`attribution_scopes`"].
+    1. If |values| is not a [=list=], return an error.
+    1. [=list/iterate|For each=] |value| of |values|:
+        1. If |value| is not a string, return an error.
+        1. If |value|'s [=string/length=] is 0 or is greater than [=max length per attribution scope=], return an error.
```

Comment:
Is there a functional reason to prohibit the empty string? If so, please add a non-normative explanation below this algorithm. Otherwise, let's allow it for simplicity.

Reply:
Now that we've merged #1235 I think we should allow the empty string here.

Reply:
+1

Reply:
Done.